src/k8s-extension/azext_k8s_extension/partner_extensions/ContainerInsights.py

```diff
@@ -550,6 +547,38 @@ def _get_container_insights_settings(cmd, cluster_resource_group_name, cluster_r
         configuration_settings['amalogs.domain'] = 'opinsights.azure.microsoft.scloud'
 
 
+def getRegionCodeForAzureRegion(cmd, cluster_region):
+    region_code = "EUS"
+    cloud_name = cmd.cli_ctx.cloud.name
+    if cloud_name.lower() == "azurecloud":
+        region_code = azurecloud_location_to_oms_region_code_map.get(
+            cluster_region, "EUS"
+        )
+    elif cloud_name.lower() == "azurechinacloud":
+        region_code = azurechina_location_to_oms_region_code_map.get(
+            cluster_region, "EAST2"
+        )
+    elif cloud_name.lower() == "azureusgovernment":
+        region_code = azurefairfax_location_to_oms_region_code_map.get(
+            cluster_region, "USGV"
+        )
+    else:
+        logger.error(
+            "AKS Monitoring addon not supported in cloud : %s", cloud_name
+        )
+    return region_code
+
+
+def sanitize_dcr_name(name):

```

Comment:
What exactly are you trying to do here?  Can you add a function header that explains this - an example would be great